### DISCUSSION TITLE
How to get the coherent chunks built by shap.plots.text

### DISCUSSION THREAD

**Discussion by MartinMayoIglesias** [shap/shap#3827] [Q&A]
the method shap.plots.text seems to be using hierarchical_values to build coherent chunks as it is mentioned in here:
https://shap.readthedocs.io/en/latest/example_notebooks/api_examples/plots/text.html

But I don't see any method that **just** returns those chunks together with the shap values of each chunk. Is it possible to do this with the current shap version?

**Comment by thatlittleboy** [comment]
Hi @MartinMayoIglesias , have a look at the `process_shap_values` function, I believe it's what you're looking for.

https://github.com/shap/shap/blob/e4e705a420b477bcdc10cdd7eec680e7e2a1da85/shap/plots/_text.py#L378

**Reply by MartinMayoIglesias** [reply]
Hi @thatlittleboy, thanks for the reply. I saw that function and I am not sure if this is what I mean. To be more precise, let me write a concrete example:

For this specific string (using the "sentiment-analysis" transformer model):

''_'i went and saw this movie last night after being coaxed to by a few friends of mine . i ' ll admit that i was reluctant to see it because from what i knew of ashton kutcher he was only able to do comedy . i was wrong . kutcher played the character of jake fischer very well , and kevin costner played ben randall with such professionalism . the sign of a good movie is that it can toy with our emotions . this one did exactly that . the entire theater ( which was sold out ) was overcome by laughter during the_'''

shap.plots.text is plotting the following 4 chunks with the following corresponded shap total values:

chunk 1: '''i went and saw this movie last night after being coaxed to by a few friends of mine . i ' ll admit that i was reluctant to see it because from what i knew of ashton kutcher he was only able to do comedy .''
shap_value: -1.93

chunk 2:'''i was wrong . kutcher played the character of jake fischer very well , and kevin costner played ben randall with such professionalism .'''
shap_value: 1.27

chunk 3: '''the sign of a good movie is that it can toy with our emotions . this one did exactly that .'''
shap_value: 4.17

chunk 4: '''the entire theater ( which was sold out ) was overcome by laughter during the'''
shap_value: 3.39

So, my question is, is there a method that can take the original text and return those chunks together with those total shap values??
The process_shap_values function you mentioned seems to be called multiple times inside shap.plots.text  and together with the fact that it returns vectors, seems to not be what I am looking for
